managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedger.java

```diff
@@ -649,4 +651,31 @@ void asyncSetProperties(Map<String, String> properties, AsyncCallbacks.UpdatePro
      * @return the future of managed ledger internal stats
      */
     CompletableFuture<ManagedLedgerInternalStats> getManagedLedgerInternalStats(boolean includeLedgerMetadata);
+
+    /**
+     * Mark deletable ledgers for bookkeeper and offload storage.
+     *
+     * @param deletableLedgerIds
+     * @param deletableOffloadedLedgerIds
+     */
+    void markDeletableLedgers(Collection<Long> deletableLedgerIds, Collection<Long> deletableOffloadedLedgerIds);
+
+    /**
+     * Get all deletable ledgers.
+     *
+     * @return all the deletable ledgers of the managed-ledger
+     */
+    Set<Long> getAllDeletableLedgers();
+
+    /**
+     * Get all deletable offloaded ledgers.
+     *
+     * @return all the deletable offloaded ledgers of the managed-ledger
+     */
+    Set<Long> getAllDeletableOffloadedLedgers();
+
+    /**
+     * Check and remove all the deletable ledgers.
+     */
+    void removeAllDeletableLedgers();
```

Comment:
Do we need to expose these methods to the ManagedLedger interface? Looks they are only used in the internal of the ManagedLedgerImpl.

Reply:
+1

Reply:
+1